src/pages/office/components/d255/d255.ts

```diff
@@ -15,23 +29,43 @@ export class D255Component implements OnChanges {
 
   @Output()
   d255Change = new EventEmitter<boolean>();
-
   private formControl: FormControl;
+  private fieldName: string = 'd255';
+
+  constructor(private outcomeBehaviourProvider: OutcomeBehaviourMapProvider) { }
 
   ngOnChanges(): void {
     if (!this.formControl) {
-      this.formControl = new FormControl(null, [Validators.required]);
-      this.formGroup.addControl('d255', this.formControl);
+      this.formControl = new FormControl(null);
+      this.formGroup.addControl(this.fieldName, this.formControl);
     }
-    if (this.d255 !== null) {
-      this.formControl.patchValue(this.d255 ? 'Yes' : 'No');
+    const visibilityType = this.outcomeBehaviourProvider.getVisibilityType(this.outcome, this.fieldName);
+
+    if (visibilityType === VisibilityType.NotVisible) {
+      this.formGroup.get(this.fieldName).clearValidators();
+    } else {
+      this.formGroup.get(this.fieldName).setValidators([Validators.required]);
     }
+
+    this.formControl.patchValue(this.getD255OrDefault());
   }
 
   d255Changed(d255FormValue: string): void {
     if (this.formControl.valid) {
-      this.d255Change.emit(d255FormValue === 'Yes' ? true : false);
+      this.d255Change.emit(d255FormValue === ValidD255Values.YES ? true : false);
```

Comment:
You can `emit` just the outcome of the expression.

Reply:
changed